botfront/imports/ui/components/nlu/models/NLUModel.jsx

```diff
@@ -337,27 +345,50 @@ class NLUModel extends React.Component {
 }
 
 NLUModel.propTypes = {
-    model: PropTypes.object.isRequired,
-    projectId: PropTypes.string.isRequired,
+    model: PropTypes.object,
+    projectId: PropTypes.string,
     instances: PropTypes.arrayOf(PropTypes.object),
     entities: PropTypes.array,
     intents: PropTypes.array,
     settings: PropTypes.object.isRequired,
-    ready: PropTypes.bool.isRequired,
+    ready: PropTypes.bool,
+    nluModelLanguages: PropTypes.array,
+    models: PropTypes.array,
+    projectDefaultLanguage: PropTypes.string,
 };
 
 NLUModel.defaultProps = {
     instances: [],
     entities: [],
     intents: [],
+    ready: false,
+    nluModelLanguages: [],
+    models: [],
+    projectDefaultLanguage: '',
+    projectId: '',
+    model: [],
 };
 
 const NLUDataLoaderContainer = withTracker((props) => {
```

Comment:
Maybe try to split up this method into smaller function
(You can put the functions directly into the method so as to not pollute the upper namespace)

Reply:
Tried to split but most of it is very inter connected. But still added some splitting of function